digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/asset/asset.scala

```diff
@@ -229,20 +233,45 @@ object ServicePointsClass {
     values.find(_.value == value).getOrElse(Unknown).isAuthorityData
   }
 
-  case object Customs extends ServicePointsClass { def value = 4;  def isAuthorityData = true;}
-  case object BorderCrossing extends ServicePointsClass { def value = 5; def isAuthorityData = true;}
-  case object RestArea extends ServicePointsClass { def value = 6;  def isAuthorityData = true;}
-  case object Airport extends ServicePointsClass { def value = 8;  def isAuthorityData = true;}
-  case object FerryTerminal extends ServicePointsClass { def value = 9;  def isAuthorityData = true;}
-  case object RailwayStation extends ServicePointsClass { def value = 11;  def isAuthorityData = true;}
-  case object ParkingArea extends ServicePointsClass { def value = 12;  def isAuthorityData = true;}
-  case object TerminalForLoadingCars extends ServicePointsClass { def value = 13;   def isAuthorityData = true;}
-  case object ParkingAreaBusesAndTrucks extends ServicePointsClass { def value = 14;   def isAuthorityData = true;}
-  case object ParkingGarage extends ServicePointsClass { def value = 15;   def isAuthorityData = true;}
-  case object BusStation extends ServicePointsClass { def value = 16;  def isAuthorityData = true;}
-  case object TaxiStation extends ServicePointsClass { def value = 10;  def isAuthorityData = false;}
-  case object ElectricCarChargingStation extends ServicePointsClass { def value = 17;  def isAuthorityData = false;}
-  case object Unknown extends ServicePointsClass { def value = 99;  def isAuthorityData = true;}
+  def apply(value: String): Int = {
+    values.find { servicePoint =>
+      val name = Normalizer.normalize(servicePoint.labelName, Normalizer.Form.NFD)
+                  .replaceAll("[^\\p{ASCII}]", "")
+                  .replaceAll("/-|\\s/g", "").toLowerCase
+      name == value
+    }.getOrElse(Unknown).value
+  }
+
+  def getTypeExtensionValue(typeExtention: String, serviceType: Int): Option[Int] = {
+    val serviceTypeClass = values.find(_.value == serviceType)
+
+    val normalizedValue = Normalizer.normalize(typeExtention, Normalizer.Form.NFD)
+      .replaceAll("[^\\p{ASCII}]", "")
```

Comment:
this Normalize + replaceAll seems to be repeated, is it possible to create a method for it?